src/include/storage/write_ahead_log/log_serializer_task.h

```diff
@@ -115,22 +131,27 @@ class LogSerializerTask : public common::DedicatedThreadTask {
   bool sleeping_ = false, empty_ = true;
 
   // Current buffer we are serializing logs to
-  BufferedLogWriter *filled_buffer_;
+  BufferedLogWriter *disk_filled_buffer_;
+  BufferedLogWriter *replication_filled_buffer_;
   // Commit callbacks for commit records currently in filled_buffer
-  std::vector<std::pair<transaction::callback_fn, void *>> commits_in_buffer_;
+  std::vector<std::pair<transaction::callback_fn, void *>> disk_commits_in_buffer_;
+  std::vector<std::pair<transaction::callback_fn, void *>> replication_commits_in_buffer_;
 
   // Used by the serializer thread to store buffers it has grabbed from the log manager
-  std::queue<RecordBufferSegment *> temp_flush_queue_;
+  std::queue<RecordBufferSegment *> temp_disk_flush_queue_;
+  std::queue<RecordBufferSegment *> temp_replication_flush_queue_;
 
   // We aggregate all transactions we serialize so we can bulk remove the from the timestamp manager
   // TODO(Gus): If we guarantee there is only one TSManager in the system, this can just be a vector. We could also pass
```

Comment:
Based on where @lmwnshn's design is going, can we make a decision about this? Would we ever want more than 1 TimestampManager in the system or was it a temporary decision for Gus' experiments?